src/BitStringAddresses.jl

```diff
@@ -784,6 +785,116 @@ function check_consistency(b::BoseFS{N,M,A}) where {N,M,A<:Union{BSAdd64,BSAdd12
   leading_zeros(b.bs.add) ≥ numBits(A) - numBits(b) ||  error("Ghost bits detected in $b.")
 end
 
+
+
+
+#################################
+"""
+    BoseFS2C{NA,NB,M,AA,AB} <: BosonicFockStateAddress <: BitStringAddressType
+
+Address type that constructed with two [`BoseFS{N,M,A}`](@ref). It represents a
+Fock state with two components, e.g. two different species of bosons with particle
+number `NA` from species A and particle number `NB` from species B. The number of
+orbitals `M` is expacted to be the same for both components.
+"""
+struct BoseFS2C{NA,NB,M,AA,AB} <: BosonicFockStateAddress
+  bsa::BoseFS{NA,M,AA}
+  bsb::BoseFS{NB,M,AB}
+end
+
+# BoseFS2C{NA,NB,M}(bsa::AA,bsb::AB) where {NA,NB,M,AA,AB} = BoseFS2C{NA,NB,M,AA,AB}(bsa,bsb) # slow - not sure why
```

Comment:
Is there a good reason to leave all this commented code in there? Otherwise it would be tidier to remove it from the PR.

Reply:
Removed.